builtin/providers/aws/resource_aws_cloudtrail.go

```diff
@@ -165,3 +190,42 @@ func resourceAwsCloudTrailDelete(d *schema.ResourceData, meta interface{}) error
 
 	return err
 }
+
+func cloudTrailGetLoggingStatus(conn *cloudtrail.CloudTrail, id string) (bool, error) {
+	GetTrailStatusOpts := &cloudtrail.GetTrailStatusInput{
+		Name: aws.String(id),
+	}
+	resp, err := conn.GetTrailStatus(GetTrailStatusOpts)
```

Comment:
I'm afraid we'll have to do explicit error checking here before the `return`, otherwise we risk crashes due to nil-dereferencing below - i.e. if there's an error, we're both trying to dereference `*resp.IsLogging` and return the error, which will likely result in crash.